tests/assembly/x86_64-xray.rs

```diff
@@ -0,0 +1,17 @@
+//@ assembly-output: emit-asm
+//@ compile-flags: -Zinstrument-xray=always -C llvm-args=-x86-asm-syntax=intel --target x86_64-unknown-linux-gnu
+//@ needs-llvm-components: x86
+//@ only-x86_64-unknown-linux-gnu
```

Comment:
But... Now we aren't actually testing this on macOS?

Reply:
Hm yeah, shouldn't this assembly smoke test be for `x86_64-apple-darwin` and `aarch64-apple-darwin` 🤔

Reply:
I don't think it's that important, because x86_64 generation is not related to mac support.

Reply:
> I don't think it's that important, because x86_64 generation is not related to mac support.

Would still be nice to have a test that it actually works on macOS.

Reply:
Anyway I added one, as well as aarch64 (but I didn't really test it because don't have device on hand)